packages/lesswrong/components/common/ContentItemBody.tsx

```diff
@@ -304,57 +289,98 @@ class ContentItemBody extends Component<ContentItemBodyProps,ContentItemBodyStat
+    const elicitBlocks = this.getElementsByClassname(element, "elicit-binary-prediction");
+    for (const elicitBlock of elicitBlocks) {
+      if (elicitBlock.dataset?.elicitId) {
+        const replacementElement = <Components.ElicitBlock questionId={elicitBlock.dataset.elicitId}/>
+        this.replaceElement(elicitBlock, replacementElement)
       }
     }
   }
 
-  hideStrawPollLoggedOut = () => {
-    const { currentUser } = this.props;
-    const { location } = this.props;
-    const { pathname } = location;
+  /**
+   * Find embedded Strawpoll blocks (an iframe integration to a polling site),
+   * and replace them with WrappedStrawPoll, which causes them to be a request
+   * to log in if you aren't logged in. See the StrawPoll block in `embedConfig`
+   * in `editorConfigs.js` (compiled into the CkEditor bundle). The DOM
+   * structure of the embed looks like:
+   *
+   *   <div class="strawpoll-embed" id="strawpoll_{pollId}>
+   *     <iframe src="https://strawppoll.com/embed/polls/{pollId}"></iframe>
+   *   </div>
+   *
+   * (FIXME: The embed-HTML in editorConfigs also has a bunch of stuff in it
+   * that's unnecessary, which is destined to get stripped out by the HTML
+   * validator)
+   */
+  wrapStrawPoll = (element: HTMLElement) => {
+    const strawpollBlocks = this.getElementsByClassname(element, "strawpoll-embed");
+    for (const strawpollBlock of strawpollBlocks) {
+      const id = strawpollBlock.getAttribute("id");
+      const iframe = strawpollBlock.getElementsByTagName("iframe");
+      const iframeSrc = iframe[0]?.getAttribute("src") ?? "";
+      const replacementElement = <Components.WrappedStrawPoll id={id} src={iframeSrc} />
+      this.replaceElement(strawpollBlock, replacementElement)
+    }
+  }
+  
+  replaceSubstrings = (element: HTMLElement) => {
+    if(this.props.replacedSubstrings) {
+      for (let str of Object.keys(this.props.replacedSubstrings)) {
+        const replacement: ContentReplacedSubstringComponent = this.props.replacedSubstrings[str]!;
 
-    if(!currentUser && this.bodyRef?.current) {
-      const strawpollBlocks = this.htmlCollectionToArray(this.bodyRef.current.getElementsByClassName("strawpoll-embed"));
-      for (const strawpollBlock of strawpollBlocks) {
-        const replacementElement = <Components.StrawPollLoggedOut pathname={pathname}/>
-        this.replaceElement(strawpollBlock, replacementElement)
+        // Find (the first instance of) the string to replace. This should be
+        // an HTML text node plus an offset into that node.
+        //
+        // We're using the dom-anchor-text-quote library for this search,
+        // which is a thin wrapper around diff-match-patch, which is a diffing
+        // library with a full suite of fuzzy matching heuristics.
+        const range: Range|null = domAnchorTextQuote.toRange(
+          element,
+          { exact: str },
+          { hint: 0 }, //TODO: store offsets with text, make use for resolving match ambiguity
```

Comment:
Does this TODO still need to be addressed in the first pass of the feature?

Reply:
Doing this would make inline-reacts more resilient to edits to the underlying comment. That'd be a nice improvement over the state quo, but isn't necessary for merging this PR (since it isn't a regression relative to master).